backend/src/v5/processors/teamspaces/projects/models/commons/tickets.groups.js

```diff
@@ -40,14 +57,23 @@ const getObjectArrayFromRules = async (teamspace, project, model, revId, rules)
 		}
 	}
 
-	const [
-		{ matched, unwanted },
-		idToMeshes,
-	] = await Promise.all([
-		getMetadataByRules(teamspace, project, model, revision, rules, { parents: 1 }),
-		getIdToMeshesMapping(teamspace, model, revision),
-	]);
+	const projection = { parents: 1,
+		...(convertTo3dRepoIds ? {} : { metadata: {
+			$elemMatch: { $or: Object.values(externalIdNamesToKeys).flat().map((n) => ({ key: n })) } } }) };
```

Comment:
probably easier to read if we have the `Object.values` part be in a different line and just put in a variable here.